Meta/Lagom/Tools/ConfigureComponents/main.cpp

```diff
@@ -192,10 +192,17 @@ static Result<Vector<DeprecatedString>, int> run_whiptail(WhiptailMode mode, Vec
         return return_code;
     }
 
-    auto file = Core::File::construct();
-    file->open(read_fd, Core::OpenMode::ReadOnly, Core::File::ShouldCloseFileDescriptor::Yes);
-    auto data = DeprecatedString::copy(file->read_all());
-    return data.split('\n');
+    auto file_or_error = Core::Stream::File::adopt_fd(read_fd, Core::Stream::OpenMode::Read);
+    if (file_or_error.is_error()) {
+        warnln("\e[31mError:\e[0m Could not adopt file descriptor for reading: {}", file_or_error.error());
+        return -1;
+    }
+    auto data_or_error = file_or_error.value()->read_all();
+    if (data_or_error.is_error()) {
+        warnln("\e[31mError:\e[0m Could not read data from file descriptor: {}", data_or_error.error());
+        return -1;
+    }
+    return DeprecatedString::copy(data_or_error.value()).split('\n');
```

Comment:
Why copy if we split after? I think `DeprecatedString::split` copies again.
(But you can also leave this since it has to be migrated from DeprecatedString anyway)

Reply:
I'd assume so, since `split()` returns a Vector of `String`, but I opted to keep it similar to the old code, only replacing the error handling. That said, I'd rather copy the data to a string first and use widely-used functions to copy and split rather than implementing an ad-hoc "split buffer and copy to strings individually" here.

Reply:
Seems perfectly reasonable